src/state.ts

```diff
@@ -58,31 +60,30 @@ const state = (new Proxy(overlay, {
 
     /** Persist sets to storage "immediately" */
     set(target, property, value) {
-        locks.withlock("state", async () => {
-            logger.debug("State changed!", property, value)
-            target[property] = value
+        logger.debug("State changed!", property, value)
+        target[property] = value
+        if (notBackground()) {
+            browser.runtime.sendMessage({type: "state", command: "stateUpdate", args: [{state: target}]})
+        } else {
+            // Do we need a global storage lock?
```

Comment:
This sounds like a good idea to me. Perhaps a promise that would perform the storage change for us and only resolve when the `storage.onChanged` event is triggered? Then we could just queue changes with `.catch(...).then(...)` and ensure that we never accidentally overwrite changes.

Reply:
Our own little library that implements this seems like a good idea.

Rather than a simple queue we could merge all incoming objects while we wait for the storage to be ready to write to to reduce our usage of it.